test/promql/tests/data_plane/pipeline_to_snapshot_test.yaml

```diff
@@ -0,0 +1,108 @@
+evaluation_interval: 1m
+
+rule_files:
+  - prometheus.pipeline_to_snapshot_alerts.yaml
+
+tests:
+
+ # Scenario when alert should NOT trigger
+  - interval: 1m
+    input_series:
+      - series: 'integration_svc_response_seconds_bucket{le="30"}'
+        values: '0+95x1'  # 95 occurrances took less than 30s
+      - series: 'integration_svc_response_seconds_bucket{le="+Inf"}'
+        values: '0+100x1'  # 100 total occurrences
+    alert_rule_test:
+      - eval_time: 1m
+        alertname: PipelineRunToSnapshotExceeded
+        exp_alerts: []
+
+ # Scenario when the alert should trigger at 15%
+  - interval: 1m
+    input_series:
+      - series: 'integration_svc_response_seconds_bucket{le="30"}'
+        values: '0+15x10'  # 15 out of 100 occurrences took less than 30s
+      - series: 'integration_svc_response_seconds_bucket{le="+Inf"}'
+        values: '0+100x10'  # 100 total occurrences
+    alert_rule_test:
+      - eval_time: 10m
+        alertname: PipelineRunToSnapshotExceeded
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              slo: true
+            exp_annotations:
+              summary: >-
+                PipelineRunFinish to SnapshotInProgress time exceeded
+              description: >-
+                Time from pipeline run finished to snapshot marked in progress has been over 30s for more than 10% of requests during the last 5 minutes
+              runbook_url: https://gitlab.cee.redhat.com/rhtap/docs/sop/-/blob/main/integration-service/pipeline_to_snapshot_exceeded.md
+
+ # Scenario when the alert should trigger at 10.1%
```

Comment:
This scenario, and also the next one, contains many metrics which are not covered by the rule. I don't think we should have those metrics.

On the other hand, the rule uses the `sum` function, but we're not testing that.

looking at this metric in staging and [production](https://grafana.app-sre.devshift.net/explore?orgId=1&left=%7B%22datasource%22:%22P22466E8E7855F1E0%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22P22466E8E7855F1E0%22%7D,%22editorMode%22:%22code%22,%22expr%22:%22integration_svc_response_seconds_bucket%7Ble%3D%5C%2230%5C%22%7D%22,%22legendFormat%22:%22__auto%22,%22range%22:true,%22instant%22:true%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D), I see that the value for `integration_svc_response_seconds_bucket{le="30"}` is generated once per cluster.

It means that this rule effectively sums over the clusters for a given environment (i.e. alerts based on the stage average and over the prod average). If that is the intention, then we should have that tested. If not, then I don't think we should use sum, or at least not in this way.

Reply:
I have written new tests that cover edge % and test across 2 clusters and removed the other buckets/inputs. Apologies if I have misunderstood your recommendations. Happy to write additional tests or change the new ones if needs be.

Reply:
I still think that the use of sum on the alert itself might not be what you're after.

A couple of questions:

1. What's the original reason for using sum here?
2. Suppose that in one cluster, all requests take forever, and on the other they all take less than 30s, I suspect we'd like to have the alert triggered. With the current implementation, this will depend on the number of requests done on each cluster (as they will be averaged).

I think the thing to do now is to understand what should be the expected behavior.

Reply:
rewritten to test with 2 clusters and different scenarios: test to verify both clusters within range, one of two out of range and two of two out of range.

Reply:
@jencull can you confirm that summing between the clusters is the expected behavior?

Reply:
@yftacherzog, each cluster will be tested individually but the metrics are aggregated then for display on grafana